neutron/live_test.go

```diff
@@ -0,0 +1,239 @@
+	gc.Suite(&LiveTests{
+		cred: cred,
+	})
+}
+
+type LiveTests struct {
+	cred     *identity.Credentials
+	client   client.AuthenticatingClient
+	neutron  *neutron.Client
+	userId   string
+	tenantId string
+}
+
+func (s *LiveTests) SetUpSuite(c *gc.C) {
+	s.client = client.NewClient(s.cred, identity.AuthUserPass, nil)
+	s.neutron = neutron.New(s.client)
+}
+
+func (s *LiveTests) TearDownSuite(c *gc.C) {
+	// noop, called by local test suite.
+}
+
+func (s *LiveTests) SetUpTest(c *gc.C) {
+	// noop, called by local test suite.
+}
+
+func (s *LiveTests) TearDownTest(c *gc.C) {
+	// noop, called by local test suite.
+}
+
+func (s *LiveTests) TestFloatingIPsV2(c *gc.C) {
+	networks, err := s.neutron.ListNetworksV2()
+	c.Assert(err, gc.IsNil)
+	var netId string
+	for _, net := range networks {
+		if net.External == true {
+			netId = net.Id
+			break
+		}
+	}
+	if netId == "" {
+		c.Errorf("no valid network to create floating IP")
+	}
+	c.Assert(netId, gc.Not(gc.Equals), "")
+	ip, err := s.neutron.AllocateFloatingIPV2(netId)
+	c.Assert(err, gc.IsNil)
+	c.Assert(ip, gc.Not(gc.IsNil))
+	c.Check(ip.IP, gc.Not(gc.Equals), "")
+	c.Check(ip.FixedIP, gc.Equals, "")
+	c.Check(ip.Id, gc.Not(gc.Equals), "")
+	c.Check(ip.FloatingNetworkId, gc.Not(gc.Equals), "")
+	ips, err := s.neutron.ListFloatingIPsV2()
+	c.Assert(err, gc.IsNil)
+	if len(ips) < 1 {
+		c.Errorf("no floating IPs found (expected at least 1)")
+	} else {
+		found := false
+		for _, i := range ips {
+			c.Check(i.IP, gc.Not(gc.Equals), "")
+			if i.Id == ip.Id {
+				c.Check(i.IP, gc.Equals, ip.IP)
+				c.Check(i.FloatingNetworkId, gc.Equals, ip.FloatingNetworkId)
+				found = true
+			}
+		}
+		if !found {
+			c.Errorf("expected to find added floating IP: %#v", ip)
+		}
+		fip, err := s.neutron.GetFloatingIPV2(ip.Id)
+		c.Assert(err, gc.IsNil)
+		c.Check(fip.Id, gc.Equals, ip.Id)
+		c.Check(fip.IP, gc.Equals, ip.IP)
+		c.Check(fip.FloatingNetworkId, gc.Equals, ip.FloatingNetworkId)
+	}
+	err = s.neutron.DeleteFloatingIPV2(ip.Id)
+	c.Assert(err, gc.IsNil)
+	_, err = s.neutron.GetFloatingIPV2(ip.Id)
+	c.Assert(err, gc.Not(gc.IsNil))
+}
+
+func (s *LiveTests) TestListNetworksV2(c *gc.C) {
+	networks, err := s.neutron.ListNetworksV2()
+	c.Assert(err, gc.IsNil)
+	for _, network := range networks {
+		c.Check(network.Id, gc.Not(gc.Equals), "")
+		c.Check(network.Name, gc.Not(gc.Equals), "")
+	}
+	firstNetwork := networks[0]
+	foundNetwork, err := s.neutron.GetNetworkV2(firstNetwork.Id)
+	c.Assert(err, gc.IsNil)
+	c.Check(foundNetwork.Id, gc.Equals, firstNetwork.Id)
+	c.Check(foundNetwork.Name, gc.Equals, firstNetwork.Name)
+}
+
+func (s *LiveTests) TestSubnetsV2(c *gc.C) {
+	subnets, err := s.neutron.ListSubnetsV2()
+	c.Assert(err, gc.IsNil)
+	for _, subnet := range subnets {
+		c.Check(subnet.Id, gc.Not(gc.Equals), "")
+		c.Check(subnet.NetworkId, gc.Not(gc.Equals), "")
+		c.Check(subnet.Name, gc.Not(gc.Equals), "")
+		c.Assert(subnet.Cidr, gc.Matches, `\d{1,3}(\.+\d{1,3}){3}\/\d+`)
```

Comment:
No. IPv4 is just what is used by the code I'm testing against.  Updating to use net.ParseCidr().  Adding IPv6 subnet to test double.